### PR TITLE
Update Example App to RN 0.62.2

### LINKED ISSUE #145: Compile Build failing for react-native 0.62 (IOS)
# Bug

When running react-native run-ios build fails.

The following build commands failed:
        CompileC /Users/rkasana/Library/Developer/Xcode/DerivedData/ZastOnmove-ghuhgjztjiovtxbhhzofecqaqfnv/Build/Intermediates.noindex/Pods.build/Debug-iphonesimulator/RNDateTimePicker.build/Objects-normal/x86_64/RNDateTimePickerManager.o /Users/rkasana/Documents/onmove/NewLennon/lennon/node_modules/@react-native-community/datetimepicker/ios/RNDateTimePickerManager.m normal x86_64 objective-c com.apple.compilers.llvm.clang.1_0.compiler

My podfile contains -
pod 'RNDateTimePicker', :path => '../node_modules/@react-native-community/datetimepicker/RNDateTimePicker.podspec'

## Environment info
System:
    OS: macOS 10.15.3
    CPU: (4) x64 Intel(R) Core(TM) i5-5350U CPU @ 1.80GHz
    Memory: 168.34 MB / 8.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 12.12.0 - /usr/local/bin/node
    Yarn: 1.19.1 - /usr/local/bin/yarn
    npm: 6.11.3 - /usr/local/bin/npm
    Watchman: 4.9.0 - /usr/local/bin/watchman
  Managers:
    CocoaPods: 1.8.4 - /usr/local/bin/pod
  SDKs:
    iOS SDK:
      Platforms: iOS 12.0, macOS 10.14, tvOS 12.0, watchOS 5.0
    Android SDK:
      API Levels: 23, 25, 27, 28, 29
      Build Tools: 28.0.3, 29.0.2
      System Images: android-29 | Google APIs Intel x86 Atom
      Android NDK: Not Found
  IDEs:
    Android Studio: Not Found
    Xcode: 10.0/10A255 - /usr/bin/xcodebuild
  Languages:
    Python: 2.7.16 - /usr/bin/python
  npmPackages:
    @react-native-community/cli: Not Found
    react: 16.11.0 => 16.11.0 
    react-native: 0.62.0 => 0.62.0 
  npmGlobalPackages:
    *react-native*: Not Found

Library version: 2.3.1

## Steps To Reproduce
1. npm install @react-native-community/datetimepicker --save
2. cd ios && pod install && cd ..
3. Tried to run react-native run-ios (Build failing and unable to run ios).

Describe what you expected to happen:

1. Project to run properly

### PR DESCRIPTION
**Pull request by luancurti**
# Summary

Update Example App to RN 0.62.2

Closes #145 

## Compatibility

| OS      | Implemented |
| ------- | :---------: |
| iOS     |    ✅❌     |
| Android |    ✅❌     |

## Checklist

- [x] I have tested this on a device and a simulator
- [ ] I added the documentation in `README.md`
- [ ] I mentioned this change in `CHANGELOG.md`
- [X] I updated the typed files (TS and Flow)
- [X] I added a sample use of the API in the example project (`example/App.js`)

### CODE REVIEW

**Comment by luancurti** [comment]
I need test the example in `Android Emulator`

**Comment by vonovak** [comment]
@luancurti just FYI `RCT_EXPORT_MODULE()` can be used without any name, just take a look here https://github.com/facebook/react-native/search?q=RCT_EXPORT_MODULE&unscoped_q=RCT_EXPORT_MODULE

but upgrading the example app is a good thing for sure 👍

**Comment by vonovak** [comment]
@luancurti hi, do you have any timeline one when you want to finish migrating the example to latest RN (=this PR)? I'm asking because I also started to work on that in https://github.com/react-native-community/datetimepicker/pull/144 but didn't finish because detox did not support 0.62.

so just asking because I want to avoid doing duplicate work :) Thanks

**Comment by luancurti** [comment]
> @luancurti hi, do you have any timeline one when you want to finish migrating the example to latest RN (=this PR)? I'm asking because I also started to work on that in #144 but didn't finish because detox did not support 0.62.
> 
> so just asking because I want to avoid doing duplicate work :) Thanks

I'll finish it today

**Comment by luancurti** [comment]
I've create an `package.json` inside `Example App` to make `auto link` works equal a real app that have `datetimepicker` with dependency

**Comment by luancurti** [comment]
We can wait the `detox` support to rn 62? @vonovak

**Comment by vonovak** [comment]
@luancurti thank you for the work!

> I've create an package.json inside Example App

I'm sorry, but I'm afraid that is not something we want to do. Could you please avoid doing that? 

also, the CI is failing (flow)

Detox can wait until we have 62 support. Thank you :)

**Comment by luancurti** [comment]
> @luancurti thank you for the work!
> 
> > I've create an package.json inside Example App
> 
> I'm sorry, but I'm afraid that is not something we want to do. Could you please avoid doing that? Why did you do the change?

I'll make the rollback. I make this change because I see another repo doing this like [react-native-share](https://github.com/react-native-community/react-native-share) but I respect your opinion

> 
> also, the CI is failing (flow)
> 
> Detox can wait until we have 62 support. Thank you :)

Ok, thank you too

File: package.json [hunk 1]
```diff
@@ -53,24 +53,25 @@
     "@babel/core": "^7.5.5",
     "@babel/runtime": "^7.5.5",
     "@react-native-community/eslint-config": "^1.0.0",
+    "@react-native-community/datetimepicker": "./",
```
Comment (vonovak) [inline package.json:56]: this looks funny, the dev dependency of `@react-native-community/datetimepicker` is `@react-native-community/datetimepicker` - why?

File: package.json [hunk 2]
```diff
@@ -53,24 +53,25 @@
...
+    "react-test-renderer": "^16.11.0"
   },
   "peerDependencies": {
-    "react": ">=16.8",
-    "react-native": ">=0.59"
+    "react": ">=16.11.0",
+    "react-native": ">=0.60"
   },
   "dependencies": {
+    "moment": "^2.24.0",
```
Comment (vonovak) [inline package.json:72]: why moving this from devDependencies to dependencies?

the dependencies block should stay the same way as it was before

File: ios/RNDateTimePickerManager.m [hunk 3]
```diff
@@ -24,7 +24,7 @@ @implementation RCTConvert(UIDatePicker)
 
 @implementation RNDateTimePickerManager
 
-RCT_EXPORT_MODULE()
+RCT_EXPORT_MODULE(RNDateTimePicker)
```
Comment (vonovak) [inline ios/RNDateTimePickerManager.m:27]: any specific reason for this change? Just wondering.

File: example/android/settings.gradle [hunk 4]
```diff
@@ -1,6 +1,6 @@
 rootProject.name = 'example'
 
 apply from: file("../../node_modules/@react-native-community/cli-platform-android/native_modules.gradle");
-applyNativeModulesSettingsGradle(settings, "../..")
+applyNativeModulesSettingsGradle(settings);
```
Comment (vonovak) [inline example/android/settings.gradle:4]: why this change? does autolinking on android still work?

File: example/android/app/build.gradle [hunk 5]
```diff
@@ -204,4 +202,4 @@ task copyDownloadableDepsToLibs(type: Copy) {
 }
 
 apply from: file("../../../node_modules/@react-native-community/cli-platform-android/native_modules.gradle");
-applyNativeModulesAppBuildGradle(project, "../..")
+applyNativeModulesAppBuildGradle(project);
```
Comment (vonovak) [inline example/android/app/build.gradle:205]: why this change? does autolinking on android still work?

Comment (luancurti) [reply example/android/app/build.gradle:205]: yes, the second argument is deprecated on rn 62 applying the `"@react-native-community/datetimepicker": "./"` in package.json the autolink works successfull

File: example/android/settings.gradle (hunk 4 above)
Comment (luancurti) [reply example/android/settings.gradle:4]: yes, the second argument is deprecated on rn 62 applying the `"@react-native-community/datetimepicker": "./"` in package.json the autolink works successfull

File: package.json (hunk 1 above)
Comment (luancurti) [reply package.json:56]: because the second paramaters is deprecated applyNativeModulesSettingsGradle(settings, "../..") I add this on package.json to autolink works equall on Android/iOS

File: ios/RNDateTimePickerManager.m (hunk 3 above)
Comment (luancurti) [reply ios/RNDateTimePickerManager.m:27]: No, I follow the docs and added this change I'll test without this and check if this still working

File: package.json (hunk 2 above)
Comment (luancurti) [reply package.json]: Done

File: ios/RNDateTimePickerManager.m (hunk 3 above)
Comment (luancurti) [reply ios/RNDateTimePickerManager.m]: Done

File: package.json (hunk 1 above)
Comment (vonovak) [reply package.json:56]: > because the second paramaters is deprecated applyNativeModulesSettingsGradle(settings, "../..")

ah, interesting, is there any resource (some docs) about this that you can point me to? Thank you!

Comment (luancurti) [reply package.json:56]: @vonovak there is https://github.com/react-native-community/cli/blob/master/packages/platform-android/native_modules.gradle#L280-L281

Comment (luancurti) [reply package.json:56]: Sorry I mention the react-native 62 but it is related to `cli`

Comment (thymikee) [reply package.json:56]: This is shouldn't be necessary. What you're expected to do, is to have `example/package.json` with a `"@react-native-community/datetimepicker": "../"` in its `dependencies`, like a normal app would do. Hope that fixes your issues. Also, remove the `--root` from `run-android`

Comment (thymikee) [reply package.json:56]: Also, you likely don't need the custom config at all. All you need to do is to run the example from the `example/` directory, like you would do for a normal app.

Check how slider does it: https://github.com/react-native-community/react-native-slider

Comment (vonovak) [reply package.json:56]: @thymikee thanks for the comment. Looking at the slider repo I see they use a very different setup with yarn workspaces it seems; there is package.json in the root, in src folder and in example folder - seems we'd need to do a bunch of refactors to use that setup and time to understand it.

If I look at a bob-generated library and example, bob also does it differently. There is package.json in the root and in the example folder - but this one does not use "some-dependency": "../" in its `dependencies` - instead `metro.config.js` in the example folder instructs metro to look for JS files in the parent folder (watchFolders: [root],).

I don't mean to complain, but this thing is really much harder than it should be.

I'll just link the native modules manually I guess.

**Comment by vonovak** [comment]
oops, I pushed directly into this branch, I didn't mean that. But anyway, I pushed those changes I wanted to make public.
@luancurti just so you know, I was doing some experiments with latest RN CLI on this branch, and pushed them in here.

**Comment by luancurti** [comment]
> oops, I pushed directly into this branch, I didn't mean that. But anyway, I pushed those changes I wanted to make public.
> @luancurti just so you know, I was doing some experiments with latest RN CLI on this branch, and pushed them in here.

Feel free to make the changes, thank for the help!

File: package.json (hunk 1 above)
Comment (luancurti) [reply package.json]: @thymikee @vonovak thanks for detailed explanation I'll take a look at the slider repository

Comment (thymikee) [reply package.json]: The slider has a different setup with Yarn workspaces, which would require lots of file shifting in this case. We identified the root cause here: https://github.com/react-native-community/cli/issues/1137#issuecomment-619988897

**Comment by luancurti** [comment]
@vonovak Awesome! 🎉  thanks again!

**Comment by vonovak** [comment]
thank you!
